### PR TITLE
feat: spdx parser

### LINKED ISSUE #623: new SBOM parser
There is a concept called SBOM (Software Bill Of Materials), and there are various tooling around this[^1][^2][^3][^4] concept to generate it. There are several methods designed specifically for delivering SBOMs, including [SPDX (Software Package Data Exchange)](https://spdx.dev), [Software Identification (SWID) Tags](https://nvd.nist.gov/products/swid), and [Cyclone DX](https://cyclonedx.org).

Wouldn't be nice if we support validating these types of SBOM specs via rego-based policies in `conftest`.

[^1]: https://github.com/anchore/syft
[^2]:  https://github.com/spdx/spdx-sbom-generator
[^3]: https://github.com/tern-tools/tern
[^4]: https://github.com/kubernetes/release/tree/master/cmd/bom

cc: @luhring @nishakm @puerco 🥳🙋🏻‍♂️

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

Fixes #623 

ezgif com-gif-maker (2)

cc: @jalseth @dentrax

### CODE REVIEW

File: parser/spdx/spdx.go
Comment (jalseth) [inline parser/spdx/spdx.go:27]: `cue` or `spdx` ?

File: parser/spdx/spdx_test.go
Comment (jalseth) [inline parser/spdx/spdx_test.go:6]: nit: This test string seems excessively long if we're only checking one field in it.

File: parser/parser.go
Comment (jalseth) [inline parser/parser.go:13]: nit: Unnecessary newline and imports should be in alphabetical order. `gofmt` should fix this automatically,

File: parser/spdx/spdx_test.go
Comment (jalseth) [inline parser/spdx/spdx_test.go:79]: nit: this error message doesn't seem to accurately reflect what is being tested

**Review by jalseth — CHANGES_REQUESTED** [review]
Hi, thanks for the PR!

In addition to the in-line comments, I'd also like to see an acceptance test added for SPDX as well before merging this.

File: parser/spdx/spdx_test.go
Comment (developer-guy) [reply parser/spdx/spdx_test.go]: thank you so much @jalseth, done.

File: parser/parser.go
Comment (developer-guy) [reply parser/parser.go]: thank you so much @jalseth, done.

File: parser/spdx/spdx_test.go
Comment (developer-guy) [reply parser/spdx/spdx_test.go:6]: thank you so much @jalseth, done.

File: parser/spdx/spdx.go
Comment (developer-guy) [reply parser/spdx/spdx.go]: thank you so much @jalseth, done.

**Comment by developer-guy** [comment]
kindly ping @jalseth

**Comment by jpreese** [comment]
This seems OK to me now, with the addition of the tests. Since @jalseth provided the initial review, he can have the final say on this 😎 

Thanks for your contribution!

File: parser/spdx/spdx_test.go
Comment (jalseth) [inline parser/spdx/spdx_test.go:31]: nit: Unit test errors should include a `have: X, want: Y` message to reduce the amount of time troubleshooting failed tests.

**Review by jalseth — CHANGES_REQUESTED** [review]
The PR is almost there. Aside from the nit, I still would like to see an example policy and acceptance test added for this new parser. See `acceptance.bats` for examples of other acceptance tests.

**Review by jalseth — CHANGES_REQUESTED** [review]
One more thing I noticed when looking over the whole parser source (not just the changes) is the SPDX was not added to the list returned by `Parsers()`: https://github.com/developer-guy/conftest/blob/ab363955507ce2eb2191bc6ccbf87e2de9a675c7/parser/parser.go#L143

File: parser/parser.go
Comment (jalseth) [inline parser/parser.go:160]: nit: Please continue ordering alphabetically as the others are.

**Review by jalseth — CHANGES_REQUESTED** [review]
Please address the nit and squash all the changes into one commit, then we can look to merge this.

File: parser/parser.go
Comment (developer-guy) [reply parser/parser.go]: done

**Comment by developer-guy** [comment]
@boranx this one also should be okay for the merge.

**Review by jalseth — APPROVED** [review]

**Comment by rnjudge** [comment]
@jalseth This is a feature I would be extremely interested in using. It's unfortunate there's no documentation or commit message information for new users like me to be able to figure out how to use or integrate this new feature. Please consider your users and new contributors in the future by requiring commit messages and some sort of documentation (even if it's only in the PR). A non-pauseable gif is not helpful in trying to understand how to use this feature.

**Pull request merged**
